src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -86,6 +89,102 @@ def _get_object_id(graph_client, subscription=None, spn=None, upn=None):
         return _get_object_id_by_upn(graph_client, upn)
     return _get_object_id_from_subscription(graph_client, subscription)
 
+
+def _get_vault_id_from_name(client, vault_name):
+    group_name = _get_resource_group_from_vault_name(vault_name)
+    vault = client.get(group_name, vault_name)
+    return vault.id
+
+
+def get_vm_format_secret(client, secrets, certificate_store=None):
+    """
+    Format secrets to be used in `az vm create --secrets`
+    :param client: management api client
+    :param dict secrets: array of secrets to be formatted
+    :param str certificate_store: certificate store the secret will be applied (Windows only)
+    :return: formatted secrets as an array
+    :rtype: list
+    """
+    grouped_secrets = {}
+    if isinstance(secrets, dict):
+        secrets = [secrets]
+
+    # group secrets by source vault
+    for secret in secrets:
+        parsed = parse_secret_id(secret['id'])
+        match = re.search('://(.+?)\.', parsed.vault)
+        vault_name = match.group(1)
+        if vault_name not in grouped_secrets:
+            grouped_secrets[vault_name] = {
+                'vaultCertificates': [],
+                'id': _get_vault_id_from_name(client, vault_name)
+            }
+
+        vault_cert = {'certificateUrl': secret['id']}
+        if certificate_store:
+            vault_cert['certificateStore'] = certificate_store
+
+        grouped_secrets[vault_name]['vaultCertificates'].append(vault_cert)
+
+    # transform the reduced map to vm format
+    formatted = [{'sourceVault': {'id': value['id']}, 'vaultCertificates': value['vaultCertificates']}
+                 for _, value in list(grouped_secrets.items())]
+
+    # This is a hack to dump out json to STDOUT with the exact formatting we desire. The issue is that
+    # this JSON contains a resource id and the default output adds a resourceGroup to the keys upon
+    # formatting for output. We don't want that to be added since it would be different from what
+    # --secrets in vm and vmss create require.
+    #
+    # Also, vm-format should only produce JSON as it's the only format that makes sense for this cmd
+    print(json.dumps(formatted, sort_keys=True, indent=2))
+    return None
+
+
+def get_default_policy(client):
+    return {
+        "attributes": {
+            "enabled": True,
+            "expires": None,
+            "not_before": None
+        },
+        "issuer_parameters": {
+            "name": "Self"
```

Comment:
Like we discussed, if a user is not doing a self-signed certificate, the command needs to show them what are the rest of the valid "issuer_parameters".  --scaffold would be one option to expose an unusable (as-is) template that would give them all the tools they will need.